src/components/ToDoForm.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+class TodoForm extends React.Component {
+  constructor() {
+    super();
+    this.newValue = {};
+  }
+
+  updateStateMessage = e => {
+    let value = e.target.value;
+
+    if (value !== null) {
+      this.newValue = { item: value, completed: false, id: Date.now() };
+ 
+    }
+  };
+
+  render() {
+    return (
+      <form onSubmit={e => {
+        
+          document.getElementById('toDoInput').value = '';
+          e.preventDefault();
+          return this.props.dispatch({
+            type: 'HANDLE_SUBMIT',
+            newValue: this.newValue
+          });
+        
```

Comment:
You can define your methods above the render method and call them on submit for cleaner code.